Proposal for non-standard NC installation

hellkaim opened the pull request:
Hello and thank you for grate script!

As I have a non-standard NextClound installation with different network port and 2FA passwords that are used to hide real username and real password from third party apps.

As such we need to use different username while connecting to the server (auth) and in the process of requesting contacts and calendar data as user names are different.

Same we need to define a port on which a NC interface is expecting connections.

I tried to make self explainable commits so it should be pretty easy to understand what I've added.

Looking forward.

hellkaim committed:
Added NC login and port

We use 2 additional parameters:
- NCLOGIN which is your login that is used for authentication. Especially needed if 2FA is enabled and you are using app passwords.
- NCPORT which is a port number of your web interface (defaults to 443 HTTPS port). Needed if your installation uses non-standard web port.

hellkaim committed:
Replacing USERNAME with NCLOGIN to incorporate difference from user and login

As you may have an AppPasswords the user could be renamed creating abstraction from a real NC user. We incorporate it in config file and now are updating Calendar and Contacts functions as they use a NC real username as endpoints rather then login username.

SaltyCybernaut commented:
Hello @hellkaim, thank you for taking the time to submit this pull request.

Can you link to any documentation or tutorial for having a "login id" that is different from your Nextcloud username?

hellkaim commented:
Hello.

By documemtation you mean official NC docs or modifying a readme od the project?

For NC you can start from here https://docs.nextcloud.com/server/latest/admin_manual/configuration_user/two_factor-auth.html

For project's README I will issue new pool request later today..

hellkaim committed:
Adding Device login-password section

As config and main script were updated to incorporate different login and Device password we need to ad correspondent info to the README.

hellkaim commented:
I have updated the README to incorporate changes and new configuration routine.

hellkaim commented:
Ok, send you an invitation. Mind the address of the web interface. I can't add anything to that account, but I can create a collection with you so we can both check the issue. Or, alternately, I can try to change passwords in my items and then we can import them somehow to that user for testing.
